custom_components/heatmiserneo/switch.py

```diff
@@ -388,4 +408,122 @@ async def async_turn_off(self, **kwargs):
         _LOGGER.info(f"{self.name} : Called set_frost with: False (response: {response})")
         self.data.standby = False
         self.async_write_ha_state()
-        
+
+
+class NeoTimerEntity(CoordinatorEntity, SwitchEntity):
+    """Represents a Heatmiser neoStat thermostat acting in TimeClock mode."""
+
+    def __init__(
+        self, neostat: NeoStat, coordinator: DataUpdateCoordinator, hub: NeoHub
+    ):
+        super().__init__(coordinator)
+        _LOGGER.debug(
+            f"Creating {type(self).__name__} for {neostat.name} {neostat.device_id}"
+        )
+
+        self._neostat = neostat
+        self._coordinator = coordinator
+        self._hub = hub
+        # Needed to filter the switches in the service call entity selector. This was the least worse option
+        self._attr_supported_features = SirenEntityFeature.DURATION
+
+    @property
+    def data(self):
+        """Helper to get the data for the current thermostat."""
+        (devices, _) = self._coordinator.data
+        neo_devices = {device.name: device for device in devices["neo_devices"]}
+        return neo_devices[self.name]
+
+    @property
+    def available(self):
+        """Return true if the entity is available."""
+        if self.data.offline:
+            return False
+        return True
+
+    @property
+    def device_class(self):
+        return SwitchDeviceClass.SWITCH
+
+    @property
+    def device_info(self):
+        return {
+            "identifiers": {
+                (
+                    DOMAIN,
+                    f"{self._coordinator.serial_number}_{self._neostat.serial_number}",
+                )
+            },
+            "name": self._neostat.name,
+            "manufacturer": "Heatmiser",
+            "model": f"{HEATMISER_PRODUCT_LIST[self.data.device_type]}",
+            "serial_number": self._neostat.serial_number,
+            "suggested_area": self._neostat.name,
+            "sw_version": self.data.stat_version,
+            "via_device": (DOMAIN, self._coordinator.serial_number),
+        }
+
+    @property
+    def extra_state_attributes(self):
+        """Return the additional state attributes."""
+        return {
+            "device_id": self._neostat.device_id,
+            "device_type": self._neostat.device_type,
+            "offline": self.data.offline,
+            "hold_on": self.data.hold_on,
+            "hold_time": ":".join(str(self.data.hold_time).split(":")[:2]),
+        }
+
+    @property
+    def name(self):
+        """Returns the name."""
+        return self._neostat.name
```

Comment:
Inconsistent naming, I know this is an open discussion point but please do not vary from the pattern, example: ```    @property
    def name(self):
        """Return the name of the sensor."""
        return f"{self._neostat.name} Heatmiser {HEATMISER_PRODUCT_LIST[self._neostat.device_type]} Temperature Hold"```

for now, I will bulk address this issue as it's own thing at a later date.

Reply:
This was just the name that was used for this switch in v1.0. I'll change it.

In terms of the general discussion about naming, this feel like the core entity of the device (turn the output on or off), so it makes sense to inherit the name from the device, in the same way as the climate entity does.

Another reason for not changing away from the stat name is that any users that were using this entity in their dashboards, automation, custom sensors etc will need to make several changes (often not very obvious)

Reply:
Updated